app/src/main/java/com/g11x/checklistapp/NavigationActivity.java

```diff
@@ -73,12 +75,19 @@ protected void onCreate(Bundle savedInstanceState) {
     final Activity thisActivity = NavigationActivity.this;
     baseLanguageChangeListener = new AppPreferences.LanguageChangeListener(NavigationActivity.this) {
       @Override
-      public void onChanged(String newValue) {
-        thisActivity.recreate();
+      void onChanged(@NonNull Language newLangauge) {
+        Intent intent = getIntent();
+        finish();
+        startActivity(intent);
```

Comment:
interesting, so my understanding was that `finish()` leads to `onDestroy()`, which may be cleaning up state that was created in `onCreate()`. and then `startActivity()` runs `onStart()`, but not necessarily `onCreate()`, which seems like it would be a problem, but i guess that `finish()` + `startActivity()` results in `onCreate()` being called? idk.